### PR TITLE
DT-1537 Translate JPA into DTO instead of calling SQL

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java [hunk 1]
```diff
@@ -154,6 +160,20 @@ public void add(final OffenderCourtCase courtCase) {
         courtCase.setOffenderBooking(this);
     }
 
+    public void add(final ProfileType profileType, final ProfileCode code) {
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java:163]: Does this have test coverage? I couldn't see any.

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import org.springframework.transaction.annotation.Transactional;
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java:10]: Are there any unit tests for this?

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java (hunk 1 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBooking.java:163]: yes, via the booking creation apis

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java (hunk 2 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java:10]: Yes in the OffendersResourceTest

**Pull request merged**